src/main.cpp

```diff
@@ -181,43 +119,72 @@ int main(int argc, char* argv[])
     }
 #endif // _MSC_VER && _DEBUG
     spDebugConsole = nullptr;
-    unsigned int startupAction = 0;
 
     // due to a Qt bug, this only safely works for both non- and HiDPI displays on 5.12+
     // 5.6 - 5.11 make the application blow up in size on non-HiDPI displays
 #if defined (Q_OS_UNIX) && (QT_VERSION >= QT_VERSION_CHECK(5, 12, 0))
     QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 #endif
 
-    QScopedPointer<QCoreApplication> initApp(createApplication(argc, argv, startupAction));
-    auto * app = qobject_cast<QApplication*>(initApp.data());
+#if defined(Q_OS_MACOS) && (QT_VERSION < QT_VERSION_CHECK(5, 12, 0))
+        // Workaround for horrible mac rendering issues once the mapper widget
+        // is open - see https://bugreports.qt.io/browse/QTBUG-41257
+        QApplication::setAttribute(Qt::AA_DontCreateNativeWidgetSiblings);
+#elif defined(Q_OS_FREEBSD)
+        // Cure for diagnostic:
+        // "Qt WebEngine seems to be initialized from a plugin. Please set
+        // Qt::AA_ShareOpenGLContexts using QCoreApplication::setAttribute
+        // before constructing QGuiApplication."
+        QApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
+#endif
+
+    auto * app = qobject_cast<QApplication*>(new QApplication(argc, argv));
```

Comment:
```suggestion
    auto app = qobject_cast<QApplication*>(new QApplication(argc, argv));
```

An improvement to the old code - we can already tell what it is from the `QApplication*` :)